okhttp/src/jvmMain/kotlin/okhttp3/internal/concurrent/TaskRunner.kt

```diff
@@ -68,7 +74,7 @@ class TaskRunner(
           } finally {
             // If the task is crashing start another thread to service the queues.
             if (!completedNormally) {
-              synchronized(this@TaskRunner) {
+              lock.withLock {
```

Comment:
Being able to avoid the hazard of dereferencing the wrong `this` is a very nice improvement